src/Http/Routing/src/DefaultLinkParser.cs

```diff
@@ -169,72 +169,46 @@ public void Deconstruct(out RoutePatternMatcher matcher, out Dictionary<string,
-                skipEnabledCheck: true);
-
-            private static readonly Action<ILogger, object, Exception> _endpointsNotFound = LoggerMessage.Define<object>(
-                LogLevel.Debug,
-                EventIds.EndpointsNotFound,
-                "No endpoints found for address {Address}");
-
-            private static readonly Action<ILogger, string, string, Exception> _pathParsingSucceeded = LoggerMessage.Define<string, string>(
-                LogLevel.Debug,
-                EventIds.PathParsingSucceeded,
-                "Path parsing succeeded for endpoint {Endpoint} and URI path {URI}",
-                skipEnabledCheck: true);
-
-            private static readonly Action<ILogger, IEnumerable<string>, string, Exception> _pathParsingFailed = LoggerMessage.Define<IEnumerable<string>, string>(
-                LogLevel.Debug,
-                EventIds.PathParsingFailed,
-                "Path parsing failed for endpoints {Endpoints} and URI path {URI}",
-                skipEnabledCheck: true);
-
-            public static void EndpointsFound(ILogger logger, object address, IEnumerable<Endpoint> endpoints)
+            public static void EndpointsFound(ILogger logger, object? address, IEnumerable<Endpoint> endpoints)
             {
                 // Checking level again to avoid allocation on the common path
                 if (logger.IsEnabled(LogLevel.Debug))
                 {
-                    _endpointsFound(logger, endpoints.Select(e => e.DisplayName), address, null);
+                    EndpointsFound(logger, endpoints.Select(e => e.DisplayName), address);
                 }
             }
 
-            public static void EndpointsNotFound(ILogger logger, object address)
-            {
-                _endpointsNotFound(logger, address, null);
-            }
+            [LoggerMessage(100, LogLevel.Debug, "Found the endpoints {Endpoints} for address {Address}", EventName = "EndpointsFound", SkipEnabledCheck = true)]
+            private static partial void EndpointsFound(ILogger logger, IEnumerable<string?> endpoints, object? address);
+
+            [LoggerMessage(101, LogLevel.Debug, "No endpoints found for address {Address}", EventName = "EndpointsNotFound")]
+            public static partial void EndpointsNotFound(ILogger logger, object? address);
 
             public static void PathParsingSucceeded(ILogger logger, PathString path, Endpoint endpoint)
             {
                 // Checking level again to avoid allocation on the common path
                 if (logger.IsEnabled(LogLevel.Debug))
                 {
-                    _pathParsingSucceeded(logger, endpoint.DisplayName, path.Value, null);
+                    PathParsingSucceeded(logger, endpoint.DisplayName, path.Value);
                 }
             }
 
+            [LoggerMessage(102, LogLevel.Debug, "Path parsing succeeded for endpoint {Endpoint} and URI path {URI}", EventName = "PathParsingSucceeded", SkipEnabledCheck = true)]
```

Comment:
These Ids conflict with the ones in DefaultLinkGenerator.Log

Reply:
@BrennanConroy would be able to confirm, but I was under the impression that the ids have to be unique within a specific logger category, not globally.

Reply:
Good point. It's not obvious from the definitions here what category they're part of, you have to backtrack to the ILogger creation.